### PR TITLE
[expo] tree-shake Logs module in production bundles

### PR DESCRIPTION
**Pull request by EvanBacon**
# Why

Most of the expo module is just providing dev logging, this PR enables tree-shaking for all the logging functionality. 
This results in a 6% reduction in total base SDK 41 bundles, but more relevantly, `expo` goes from being 4.2% of a base bundle to being 0.8%.

### Before

Screen Shot 2021-04-12 at 10 09 24 PM

### After

Screen Shot 2021-04-12 at 10 09 30 PM

### CODE REVIEW

File: packages/expo/src/logs/Logs.ts [hunk 1]
```diff
@@ -1,21 +1,24 @@
-import RemoteConsole from './RemoteConsole';
-
 let _originalConsole: typeof console | null;
 
 export function enableExpoCliLogging(): void {
-  if (_originalConsole) {
-    return;
-  }
+  if (__DEV__) {
```
Comment (ide) [inline packages/expo/src/logs/Logs.ts:4]: Is this necessary with the main DEV block?

Comment (EvanBacon) [reply packages/expo/src/logs/Logs.ts:4]: yes, the other dev block is a side effect, this dev block is for the re-export through the main expo module.

**Review by ide — APPROVED** [review]

**Review by tsapeta — APPROVED** [review]

**Comment by ide** [comment]
Make sure to fix the tests before landing.

File: packages/expo/build/environment/logging.fx.js [hunk 2]
```diff
@@ -1,16 +1,19 @@
...
-    // remote debugger). In Expo Web we don't show console logs in the CLI, so there's no special case needed.
-    if (!isRunningInWebBrowser()) {
-        Logs.enableExpoCliLogging();
-    }
-    else {
-        RemoteLogging.enqueueRemoteLogAsync('info', {}, [
-            'You are now debugging remotely; check your browser console for your application logs.',
-        ]);
+// Metro and terser don't seem to be capable of shaking the imports unless they're wrapped in __DEV__.
+if (__DEV__) {
```
Comment (shamilovtim) [inline packages/expo/build/environment/logging.fx.js:3]: @EvanBacon How are you guys using `__DEV__` to treeshake Metro? Even using the `require()` API with __DEV__ still doesn't treeshake Metro in our project with `metro-babel-preset`.

**Pull request merged**